src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogSink.cs

```diff
@@ -93,42 +126,57 @@ protected override async Task EmitBatchAsync(IEnumerable<LogEvent> events)
         /// the object is being disposed from the finalizer.</param>
         protected override void Dispose(bool disposing)
         {
-            _client.Close();
-            base.Dispose(disposing);
-        }
-
-        private static IDatadogClient CreateDatadogClient(string apiKey, string source, string service, string host, string[] tags, DatadogConfiguration configuration, bool detectTCPDisconnection)
-        {
-            var logFormatter = new LogFormatter(source, service, host, tags);
-            if (configuration.UseTCP)
+            if (disposing)
             {
-                return new DatadogTcpClient(configuration, logFormatter, apiKey, detectTCPDisconnection);
-            }
-            else
-            {
-                return new DatadogHttpClient(configuration, logFormatter, apiKey);
+                try
+                {
+                    // delay the dispose by one batch period so lingering events get logged. 
+                    Task.Delay(DefaultBatchPeriod, _cancellationToken).Wait(_cancellationToken);
```

Comment:
The reason for this delay is that if you use the sink as shown in README.md:
```
using (var log = new LoggerConfiguration()
    .WriteTo.DatadogLogs("<API_KEY>")
    .CreateLogger())
{
    log.Information("Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
}
```

The call to ` log.Information` will complete immediately and the using block will exit which raises `Dispose()` - because we now clean up the sink resources and flow the `CancellationToken`, the event passed to `log.Information` will never reach the wire. You're basically writing to `/dev/null` at that point.

So, to avoid breaking existing implementations _and_ to allow the sink a chance to write pending batches this block makes sense. Users also shouldn't be constantly disposing the sink over-and-over so this block is really only ever going to be called at the end of the application. ` CloseAndFlush` also does not pump any pending events to the `WriteAsync` method, so really this block is purely about giving things in-flight a chance to finish.